### DISCUSSION TITLE
autocorr-times and acceptance-fractions

### DISCUSSION THREAD

**Discussion by SamiraMetanat** [phoebe-project/phoebe2#578] [Questions]
Hello,
We are working on 2 contact systems. We run the MCMC with interaction 1000 for each of them. The corner plot is not what we think is acceptable and we checked the autocorr-times and acceptance-fractions and they are going to radius with more interactions. We don't know what part of our work has the problem! One of them has no the autocorr-times!

![Star3](https://user-images.githubusercontent.com/102252866/159770791-7e41a6ba-97cf-4f54-9e7a-fb294e1a207a.jpg)
![Star3-](https://user-images.githubusercontent.com/102252866/159770808-061323f2-dfbd-4130-ad63-74700d71b80d.jpg)
![V336TrA-mcmc_10-phase](https://user-images.githubusercontent.com/102252866/159770813-47f879b0-d8b2-46fc-a3c8-d2fc10f98356.jpg)
![IMG-20220323-WA0018](https://user-images.githubusercontent.com/102252866/159770860-06c5fef4-4eeb-4758-be60-005bd16d52d5.jpg)

**Comment by SamiraMetanat** [comment]
iteration is 1000 and walkers is 32.

**Reply by gecheline** [reply]
Hi,

There are several things you can do to improve the corner plots, based on what I'm seeing in the log probability (don't worry about the autocorrelation times for now, you can revisit them with the autocorrelation plots on the cleaned samples).

There are two parameters that determine which samples in MCMC are used for the corner plot: burnin - number of initial iterations to discard, and lnprob_cutoff - discard samples with a logprobability value lower than the cutoff. I see that there is one branch in the logprob plot that is pretty flat from around 200 iterations onward, and another one separated by a gap in logprob values that slowly converges towards the top one. So try to remake the corner plot with passing burnin=200 and lnprob_cutoff = 1150 (you can probably get a better value for the cutoff if you zoom in on the plot near that top branch).

The other thing I notice is that you have a bimodal inclination posterior because you're sampling mirrored values around 90. I usually fix this by manually mirroring them back on a range [0,90] with:

`b['samples@mcmc_solution'].value[:,:,0][b['samples@mcmc_solution'].value[:,:,0] > 90] = 180 - b['samples@mcmc_solution'].value[:,:,0][b['samples@mcmc_solution'].value[:,:,0] > 90]`

You would need to change @mcmc_solution in the line above to the label of your mcmc solution in the bundle. @kecnry I think there should have been some functionality for this that handles things under the hood or am I wrong?

**Reply by kecnry** [reply]
PHOEBE will automatically provide a "prior" to ensure the sampler doesn't wander more than a full cycle away from the original values, but won't handle the mirroring here.  You could restrict them from going past 90 by setting a uniform prior in advance, or by editing the samples values after... but depending on whether the "tails" of the distribution hits that cutoff, it may no longer be a safe assumption to interpret it as a guassian distribution.

**Reply by gecheline** [reply]
That is good enough. I've noticed that if you provide a hard bound at 90 with a uniform prior it can artificially force your value away from 90, so I wouldn't advise that if the mean + standard deviation hit 90. And I guess my suggestion is a suitable way to go here since they already have the results.

**Comment by SamiraMetanat** [comment]
Thank you so much. All clear.
